### PR TITLE
[Merged by Bors] - Add GCP Kubernetes support

### LINKED ISSUE #168: Add GKE support
If your kubeconfig includes a GKE cluster, the config parsing will fail due to this currently unsupported `auth_provider` block within the users.

```
 auth-provider:
      config:
        access-token: xxx
        cmd-args: config config-helper --format=json
        cmd-path: /opt/homebrew/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/bin/gcloud
        expiry: "2022-05-10T00:11:22Z"
        expiry-key: '{.credential.token_expiry}'
        token-key: '{.credential.access_token}'
```

Additionally, access is primarily dynamic token based, not cert based. So client requests need to be modified to identify when a token is not static and how to renew it.

For simplicity, I think it is a good enough start to assume `gcloud` is installed, and it was used to write cluster info into the config via `gcloud container clusters get-credentials`. This way we can use the `cmd-path` + `cmd-args` to do the token retrieval.

### PR DESCRIPTION
**Pull request by tjtelan**
Resolves #168 

Fixes panic during kubernetes config parse due to cluster use with auth provider and dynamic tokens (like in GCP)
  - Only support for token refresh support via exec of CLI command in auth provider named `gcp`

Add support for GCP kubernetes (GKE) clusters
  - Tested w/ Autopilot and Standard clusters

### CODE REVIEW

**Comment by sehz** [comment]
Can you assign to issue?

File: src/k8-client/src/cert.rs [hunk 1]
```diff
@@ -78,6 +80,57 @@ where
...
+                .contexts
+                .iter()
+                .find(|context| context.name == current_context)
+            {
+                let users = &kube_config.users;
+
+                let token = if let Some(u) = users.iter().find(|user| user.name == c.context.user) {
+                    if let Some(auth_provider) = &u.user.auth_provider {
+                        if let AuthProviderDetail::Gcp(gcp_auth) = auth_provider {
+                            debug!("{gcp_auth:#?}");
```
Comment (sehz) [inline src/k8-client/src/cert.rs:100]: can you ensure that secrete keys and other auth is not logged?

File: src/k8-client/src/cert.rs [hunk 2]
```diff
@@ -78,6 +80,57 @@ where
...
+
+                            // Execute the command by default just in case access_key is expired
+                            let output = std::process::Command::new(&gcp_auth.cmd_path)
+                                .args(gcp_auth.cmd_args.split_whitespace().collect::<Vec<&str>>())
+                                .output()
+                                .expect("gcp command failed");
```
Comment (sehz) [inline src/k8-client/src/cert.rs:106]: turn into error instead of panicking

File: src/k8-client/src/cert.rs [hunk 3]
```diff
@@ -78,6 +80,57 @@ where
...
+
+                            // Return token from json response
+                            if let Ok(json) = serde_json::from_slice::<Value>(&output.stdout) {
+                                debug!("{json:#?}");
```
Comment (sehz) [inline src/k8-client/src/cert.rs:110]: ensure secrets are not logged

File: src/k8-client/src/cert.rs [hunk 4]
```diff
@@ -78,6 +80,57 @@ where
...
+                                debug!("{:#?}", json["credential"]["access_token"]);
+
+                                json["credential"]["access_token"]
+                                    .as_str()
+                                    .map(String::from)
+                            } else {
+                                None
+                            }
+                        } else {
+                            error!("Only Auth provider support for Google Kubernetes Engine (GKE). Please file issue.");
```
Comment (sehz) [inline src/k8-client/src/cert.rs:120]: this should return error

File: src/k8-client/src/cert.rs [hunk 5]
```diff
@@ -78,6 +80,57 @@ where
...
+
+            if let Some(c) = &kube_config
+                .contexts
+                .iter()
+                .find(|context| context.name == current_context)
+            {
+                let users = &kube_config.users;
+
+                let token = if let Some(u) = users.iter().find(|user| user.name == c.context.user) {
+                    if let Some(auth_provider) = &u.user.auth_provider {
```
Comment (sehz) [inline src/k8-client/src/cert.rs:98]: GCP specific token execution should move to `AuthProviderDetail`?

**Review by sehz — CHANGES_REQUESTED** [review]

**Comment by tjtelan** [comment]
The most recent commit isn't compatible w/ fluvio, so I wouldn't merge this yet until that get sorted out

**Review by sehz — APPROVED** [review]
LGTM

**Pull request closed without merging**